README.md

```diff
@@ -64,61 +98,116 @@ webhook matches the value of `co.elastic.traces/agent` (in this case, `java`) to
 config with the matching name under `webhookConfig.agents` defined in the
 helmchart.
 
-Custom webhook configurations can be defined. For example, editing `custom.yaml`:
-
-```
-webhookConfig:
-  agents:
-    prodJava:
-      image: docker.elastic.co/observability/apm-agent-java:1.30.1
-      artifact: "/usr/agent/elastic-apm-agent.jar"
-      environment:
-        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar"
-        ELASTIC_APM_SERVER_URL: "https://10.10.10.10:8200"
-        ELASTIC_APM_ENVIRONMENT: "prod"
-        ELASTIC_APM_LOG_LEVEL: "info"
-    devJava:
-      image: docker.elastic.co/observability/apm-agent-java:1.30.1
-      artifact: "/usr/agent/elastic-apm-agent.jar"
-      environment:
-        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar"
-        ELASTIC_APM_SERVER_URL: "https://10.10.10.10:8200"
-        ELASTIC_APM_ENVIRONMENT: "dev"
-        ELASTIC_APM_LOG_LEVEL: "debug"
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-service
+  namespace: default
+  labels:
+    app: my-service
+    service: my-service
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-service
+  template:
+    metadata:
+
+      # APM Mutating WebHook configuration
+      annotations:
+        co.elastic.traces/agent: java
+
+      labels:
+        app: my-service-java
+        service: my-service
+    spec:
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+      - name: my-service
+        image: my-service:v1.0.0
+        ports:
+        - name: my-service
+          containerPort: 8080
 ```
 
-Note: `artifact` and `JAVA_TOOL_OPTIONS` keys should not be edited.
-
-To use the custom config when installing the webhook, supply `-f custom.yaml`
-to the `helm upgrade` above.
-
-A deployment can indicate which configuration to apply via its annotation, ie.
-`co.elastic.traces/agent: devJava`.
-
-# configuring
-
-The user can (and should) pass in a custom yml config on creation.
+Using the annotation value allows users to set custom environment variables and
+images per deploy. For example, `backend1` might have a different service name
+from `backend2`, and `backend1-dev` might have a different apm environment from
+`backend1-prod`.
 
 ```yml
 agents:
-  java:
-    image: docker.com/elastic/agent-java:1.2.3
+  backend1:
+    image: docker.elastic.co/observability/apm-agent-java:1.33.0
     artifact: "/usr/agent/elastic-apm-agent.jar"
     environment:
-      ELASTIC_APM_SERVER_URLS: "http://34.78.173.219:8200"
+      ELASTIC_APM_SERVER_URLS: "http://192.168.1.10:8200"
       ELASTIC_APM_SERVICE_NAME: "petclinic"
```

Comment:
I'd remove the service name config form the general agent declaration as this is going to differ for each application that uses the config. The idea is to rely on the auto-detection of the service name. If that's not sufficient, users can also add the `environment` directly to their pod config that's specific for that pod.

```suggestion
```